fluidefi/fluidefi-caspernet-aggregator-tools
# Event Mapper Service

**Summary of change:**
- Added a new micro-service, deploys finder, which accepts block number as an argument and returns the deploy. 
- Updated deploys service entity to include `deployIndex`.
- Added a new micro-service, deploys parser, which accepts deploy fields as an arguments and returns parsed deploys.
- Added a new micro-service, event filter, which accepts parsed fields as an argument and filters events according to type.
- Added parser and saver micro-service for all types of events. 

**List of tasks:** 
- Deploy Finder

- Deploys
    - [x] updated entity.
    - [x] updated queries. 
    - [x] updated interface.
    - [x] updated types.
    - [x] updated class.

- Event Parser
    - [x] created interface.
    - [x] created types.
    - [x] created class.
    - [x] created example.
    - [x] created unit test.

- Event Filter
    - [x] created fixtures.
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created unit test.

- PairCreatedEventParser
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created unit test.

- PairCreatedEventSaver
    - [x] created entity.
    - [x] created query.
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created mocks.
    - [x] created fixtures.
    - [x] created unit test.

- ERC20TransferEventParser
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created unit test.

- ERC20TransferEvent Saver
    - [x] created entity.
    - [x] created query.
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created mocks.
    - [x] created fixtures.
    - [x] created unit test.

- ERC20ApprovalEventParser
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created unit test.

- ERC20ApprovalEvent Saver
    - [x] created entity.
    - [x] created query.
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created mocks.
    - [x] created fixtures.
    - [x] created unit test.

- SwapEventParser
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created unit test.

- SwapEvent Saver
    - [x] created entity.
    - [x] created query.
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created mocks.
    - [x] created fixtures.
    - [x] created unit test.

- BurnEvent Parser
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created unit test.

- BurnEvent Saver
    - [x] created entity.
    - [x] created query.
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created mocks.
    - [x] created fixtures.
    - [x] created unit test.

- SwapEvent Parser
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created unit test.

- SwapEvent Saver
    - [x] created entity.
    - [x] created query.
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created mocks.
    - [x] created fixtures.
    - [x] created unit test.

- MintEvent Parser
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created unit test.

- MintEvent Saver
    - [x] created entity.
    - [x] created query.
    - [x] created types.
    - [x] created interface.
    - [x] created class.
    - [x] created example.
    - [x] created mocks.
    - [x] created fixtures.
    - [x] created unit test.

**Steps for testing**
- Make sure your local git is up to date
```bash
git pull
```

- Checkout the branch `feature/event-mapper`
```bash
git checkout feature/event-mapper
```

- Run the test across the test-net
```bash
npm run test
```

- Confirm all tests have passed.
